cfep-23.md

```diff
@@ -0,0 +1,74 @@
+
+The integration in the Element ecosystem was seen as exciting and positive: emoji reactions were available for the first time, and threads were shortly implemented too. However, over time the excitement faded away, which prompted the proposal discussed in this CFEP.
+
+## Motivation
+
+Element/Matrix offers a free to use real-time chat platform with web, desktop and mobile clients with end-to-end encryption messaging. Each community or space can create one or more rooms, public or private, to discuss different topics. Other features like videocalls, private messaging, and file sharing are available. Users can create a new account or log in with their Github credentials, among others.
+
+However, it suffers from a few issues that limit its usability within our community. The main concerns are:
+
+- End-to-end encryption fails sometimes, leaving the users unable to read previous messages, or have their messaged read by others. Synchronization of keys across clients is necessary to fix it, but in some other cases the issue has dissapeared spontaneously.
+- Some messages are considered unread for months at a time, and there's no way to find them and mark them as read.
+- The thread system offers very limited ways of tracking conversations and making them discoverable. It's very difficult to follow on topics started hours ago, let alone days or weeks ago.
+- Scoped discussions require a lot of discipline from the users, since threads are optional, and the real-time chat is preferred in the UI.
+
+Zulip, on the other side, is a free to use, open source, real-time discussion board. It has web, desktop and mobile clients. Each community has its Zulip instance, where several "channels" can be created. Each channel hosts a number of threads or topics. Instant messaging is also available within the Zulip instance. Users can also create a new account or log in with their Github credentials, among others.
+
+Zulip offers some features we would consider beneficial to the conda-forge community:
+
+- Threads (or topics) are encouraged and the default user experience.
+- Each topic must have a title that promotes well-scoped discussions, which can be marked as solved when the issue has been addressed.
+- Moderation tools are available to split threads into new ones in case the conversation goes off-topic.
+- Users can "follow" threads they find interesting, which has its own dedicated view. This makes it easier to track updates to conversations you have started or participated in.
+- Private rooms are also available, but they don't offer end-to-end encryption. For this matter, a sensitive communications protocol is proposed below.
+
+## Proposal
+
+After accepting this proposal, we will implement the following decisions:
+
+- The Zulip instance published at `conda-forge.zulipchat.com` will be open to the public with world-readable permissions.
```

Comment:
I think one thing we could think about is wether we want to merge with `conda-community`. I believe having multiple zulip's is a little more friction than multiple element's (because you have to switch subdomains). 

I think I would prefer those two to live close to each other.

Reply:
I agree with this.

Reply:
I'm not against this but there are some concerns that would need to be resolved first:

1. Room permissions:
  - The conda community and conda-forge are two separate governed bodies
  - That implies that they will have their separate private rooms to discuss their steering council / core matters
  - Admins will have access to all rooms
  - Hence only members of both governing bodies would be able to have admin permissions. The rest of the teams would need to be moderators and access would need to be restricted. This need to be encoded in the policy.
2. We will need to move the conda community there first, which will require its own CEP, and only then modify this CFEP to vote for our transition to the conda zulip. 
  - Both the CEP and CFEP would have to same the provisions about access control and membership levels. 
  - This would require that we will always need at least one person in both governing bodies. Right now we have (I think) 11 members in common, but who knows how would this look in the future.

Again nothing impossible, but bureaucratically more involved.

Reply:
While I'd love to have simpler communication channels for the communities I'm in. I'm afraid this is not a good idea. NumFOCUS review each org independently from time to time and imagine this scenario: One community is no longer a sponsor project but it is still part of a sponsored community communication infrastructure. How would we manage that? Depending on the reasons for the removal of sponsor status, we may be force to "kick out" one of them from zulip.

I don't want to open a can of worms on what-ifs, b/c there are many other fronts where this would be problematic. Also, that scenario is highly unlikely, still, I'd rather not be in that situation.

Reply:
In any case this is not a sticking point for me so happy to go ahead with two spaces.

Reply:
I installed the Zulip Desktop app and it's OK to have two zulips side-by-side that way. Not a blocker from me either, just think the conda community is small enough that it would make sense to have a single spot. But maybe we'll grow!

Reply:
> This is not true AFAIK. I had to be invited to the steering room for conda even though I am an admin.

@beckermr Can you see a new room named `private-test` from the "Browse / add channels" dialog? Click on this icon:

<img width="359" alt="image" src="https://github.com/user-attachments/assets/d7733a7e-daf8-45fe-97b4-fc5275255be1">

Reply:
I can see it but I cannot join. 

![image](https://github.com/user-attachments/assets/0b3acf05-81d3-4484-bf13-c78afa0dd72e)

Reply:
Perfect, thanks for the clarification. I made you an Owner now (apparently it's a different level), does that make a difference @beckermr?

Reply:
No it does not. :/

Reply:
Jaime think this is a room setting issue. Looking at the description it appears new subscribers can only be added by existing subscribers

<img width="1212" alt="Screenshot 2024-09-25 at 1 55 33 PM" src="https://github.com/user-attachments/assets/e7ae7866-f251-4d68-a1cd-5237b25590c2">

<hr>

Separately what is the distinction between Administrators and Owners? Should we just all be made Owners?

Reply:
Hm, ok, good to know. @beckermr @jakirkham, can you see these buttons in the top part of that dialog that would allow you to delete a channel?

<img width="656" alt="image" src="https://github.com/user-attachments/assets/fd5fcb9c-3950-4e70-949f-619ed4cd21ee">

Reply:
yes I can see the trashcan and it appears that I could click on it, though I did not.

Reply:
Yes I can see the trashcan, but the subscribe button is still gray

Am unclear whether the latter is fixable. It kind of seems like a Zulip bug atm

<img width="1212" alt="Screenshot 2024-09-26 at 1 41 55 PM" src="https://github.com/user-attachments/assets/ca190daa-e83b-41e3-886d-2aea9fb1c4f7">

<hr>

> > Should we just all be made Owners?
>
> Yes, I don't think there's any reason not to. Right now we are not because... I just learnt about that role distinction as part of this conversation 😬 

All good. We are all learning 😅

Reply:
It does look while admins and owners cannot see private messages unless explicitly invited by already subscribed users (which is fair), owners can indeed delete the private group chats altogether. Maybe by design, but I don't think it's a bug.

Reply:
As a result, my concerns of cross-organization visibility are not that severe (owners can't peek!) but they are equally destructive (owners can delete). So if we ever get to a merged conda + conda-forge Zulip community, only members of both entities could be owners.

But since it does look like having two separate instances is not the end of the world, I think we can close this thread here.